tools/wptrunner/wptrunner/wptcommandline.py

```diff
@@ -551,6 +551,9 @@ def check_args(kwargs):
     if kwargs["processes"] is None:
         kwargs["processes"] = 1
 
+    if kwargs["device_serial"]:
+        kwargs["processes"] = len(kwargs["device_serial"])
```

Comment:
If `--device-serial` overrides `--processes`, should it be an error to specify both? Whatever behavior you go with, can you add a comment here explaining why it needs to be this way? Otherwise, a reader like me would wonder *why* it's not possible to run multiple emulators but just one process, or even vice versa.

Reply:
> If --device-serial overrides --processes, should it be an error to specify both?
Yes, let me update the code to make sure they are not specified at the same time. But that will only serve as an alert to the user, telling them they should not specify both. We will still need the code here, to correctly setup number of processes to use.

> why it's not possible to run multiple emulators but just one process
I thought under current architecture, one process is intended to run with one browser instance? An emulator + the browser running inside it is something equivalent to the browser instance.

> or even vice versa
This means we will run multiple processes with one emulator, and we will launch multiple browser instances inside the same emulator. I am not very familiar with chromedriver, so not sure if it can launch multiple browser instances, and interact with them correctly. Also I think it is a question mark if we can launch multiple browser instances inside same emulator.
But of course if we can find some more efficient way to do this, we should do that. So maybe I should talk to chromedriver team to check with them? I did find when run multiple emulators, they are also multiple chromedriver instances.

Let me add some comments here.

Reply:
FWIW I'm OK with the idea that there's one device per process on android. I don't think we can run multiple copies of the browser under test on a single android device, and at least geckodriver can only control one browser per instance. That's also the model we use for desktop, and trying to multiplex multiple browsers through a single driver would be a pretty substantial refactor.

In terms of the command line arguments, I think:
* If you don't specify `--processes` but do specify `--device-serial` we should update the number of processes to match the number of devices (and log that we're doing so).
* If you specify `--processes` and `--device-serial` and the numbers match, we should proceed without error.
* If you specify both but the numbers don't match, we should fail.

Reply:
Thanks. I updated this pr per jgraham's suggestion. Please review again.